src/lib.rs

```diff
@@ -11,28 +11,214 @@ use rand::Rng;
 use rustty::ui::Widget;
 use rustty::CellAccessor;
 
+#[macro_use]
+pub mod macros {
+    #[macro_export]
+    macro_rules! ovec { ($($elem:expr),* $(,)*) => { vec![$($elem.to_owned()),*] } }
+}
+
 type Cell = (usize, usize);
 type CellSet = HashSet<Cell>;
 
+
+#[derive(PartialEq, Eq, Debug)]
+pub struct Grid {
+    grid: CellSet,
+    bounds: Option<(usize, usize)>,
+}
+
+impl Grid {
+    fn new(b: Option<(usize, usize)>) -> Self {
+        match b {
+            None => Grid {
+                grid: CellSet::new(),
+                bounds: b,
+            },
+            _ => Grid {
+                grid: CellSet::with_capacity(b.unwrap().0 * b.unwrap().1),
+                bounds: b,
+            },
+        }
+
+    }
+    fn insert(&mut self, cell: Cell) {
+        if self.bounds != None {
+            if cell.0 <= self.bounds.unwrap().0 && cell.1 <= self.bounds.unwrap().1 {
+                self.grid.insert(cell);
+            }
+        } else {
+            self.grid.insert(cell);
+        }
+    }
+    fn contains(&self, cell: &Cell) -> bool {
+        self.grid.contains(cell)
+    }
+
+    fn x_bound(&self) -> usize {
+        self.bounds.unwrap().0
+    }
+
+    fn y_bound(&self) -> usize {
+        self.bounds.unwrap().1
+    }
+
+    pub fn gen(&mut self) {
+        if self.bounds == None {
+            return;
+        }
+        self.grid.clear();
+        for x in 0..self.bounds.unwrap().0 {
+            for y in 0..self.bounds.unwrap().1 {
+                if rand::thread_rng().gen_weighted_bool(10) {
+                    self.insert((x, y));
+                }
+            }
+        }
+    }
+}
+
+impl From<Vec<String>> for Grid {
+    /// Returns a Grid interpreted from a string representation
+    ///
+    /// # Arguments
+    ///
+    /// * `s` - Representation of the grid. Each element of the vector
+    /// represents a row in the grid. Hash marks # indicate live cells.
+    /// For example ovec!["#  ", "   ", " # "] represents a grid with live
+    /// cells at (0, 0) and (2, 1).
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # #[macro_use] extern crate hemoglobin; fn main(){
+    ///
+    /// let grid = hemoglobin::Grid::from(ovec!["#  ", "   ", " # "]);
+    /// # }
+    /// ```
+    fn from(s: Vec<String>) -> Self {
+        let mut result = Grid::new(None);
```

Comment:
A boundless `Grid` is there because our choice of a HashMap effectively allows for an infinite grid to be used. Right now it's not supported, but since the data types do support it I think we should move into making infinite worlds a possibility.

Reply:
Here I'm not sure whether to establish that every `Grid` from strings is boundless, and then add a function to establish bounds; or to infer a boundary and add a function to remove them. Thoughts?

Reply:
The cell x- and y-coordinates are `usize`, so definitely not infinite (and you don't lose anything by making the `Grid` bounds required `usize`).

Reply:
I'm opening an issue to discuss this further :)